src/OnlineSales/Controllers/ContactsController.cs

```diff
@@ -10,15 +10,64 @@
 using OnlineSales.Data;
 using OnlineSales.DTOs;
 using OnlineSales.Entities;
+using OnlineSales.Interfaces;
 
 namespace OnlineSales.Controllers;
 
 [Authorize]
 [Route("api/[controller]")]
 public class ContactsController : BaseControllerWithImport<Contact, ContactCreateDto, ContactUpdateDto, ContactDetailsDto, ContactImportDto>
 {
-    public ContactsController(ApiDbContext dbContext, IMapper mapper, IOptions<ApiSettingsConfig> apiSettingsConfig)
+    private readonly IContactService contactService;
+
+    public ContactsController(ApiDbContext dbContext, IMapper mapper, IOptions<ApiSettingsConfig> apiSettingsConfig, IContactService contactService)
         : base(dbContext, mapper, apiSettingsConfig)
     {
+        this.contactService = contactService;
+    }
+
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    public override async Task<ActionResult<ContactDetailsDto>> Post([FromBody] ContactCreateDto value)
+    {
+        var contact = mapper.Map<Contact>(value);
+
+        var savedContact = await contactService.AddContact(contact);
+
+        var returnedValue = mapper.Map<ContactDetailsDto>(savedContact);
+
+        return CreatedAtAction(nameof(GetOne), new { id = savedContact.Id }, returnedValue);
+    }
+
+    [HttpPatch("{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status422UnprocessableEntity)]
+    public override async Task<ActionResult<ContactDetailsDto>> Patch(int id, [FromBody] ContactUpdateDto value)
```

Comment:
@WASandaruwan we should also have HTTP 500 for all the controller method as some unexpected errors may happen in any part of the code, so HTTP 500 should be added for any API method we have.